src/core/server/doc_links/doc_links_service.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getDocLinks, getDocLinksMeta } from '@kbn/doc-links';
+import type { CoreContext } from '../core_context';
+import type { DocLinksServiceSetup, DocLinksServiceStart } from './types';
+
+/** @internal */
+export class DocLinksService {
+  private readonly kibanaBranch: string;
+  private docLinks?: DocLinksServiceSetup;
+
+  constructor(core: CoreContext) {
+    this.kibanaBranch = core.env.packageInfo.branch;
+  }
+
+  public setup(): DocLinksServiceSetup {
+    const docMeta = getDocLinksMeta({ kibanaBranch: this.kibanaBranch });
+    const docLinks = getDocLinks({ kibanaBranch: this.kibanaBranch });
+    this.docLinks = {
+      ...docMeta,
+      links: docLinks,
+    };
+    return this.docLinks;
```

Comment:
On the client, we're only exposing the docLinks from the start contract, however on the server-side, I think there's some scenarios where consumers will need to access it at setup too, so I exposed it from both lifecycles.